backup_and_restore/application_backup_and_restore/oadp-api.adoc

```diff
@@ -0,0 +1,260 @@
+|Defines the configuration of the `Restic` pod.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#ResticConfig[Complete schema definitions for the type `ResticConfig`].
+
+.PodConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`nodeSelector`
+|map [ link:https://pkg.go.dev/builtin#string[string] ] link:https://pkg.go.dev/builtin#string[string]
+|Defines the `nodeSelector` to be supplied to a `Velero` `podSpec` or a `Restic` `podSpec`.
+
+|`tolerations`
+|[]link:https://pkg.go.dev/k8s.io/api/core/v1#Toleration[Toleration]
+|Defines the list of tolerations to be applied to a Velero deployment or a Restic `daemonset`.
+
+|`resourceAllocations`
+|link:https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements[ResourceRequirements]
+|Set specific resource `limits` and `requests` for a `Velero` pod or a `Restic` pod as described in xref:../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc#oadp-setting-resource-limits-and-requests_installing-oadp-aws[Setting Velero CPU and memory resource allocations].
+
+If a `Velero` pod or a `Restic` pod fails or restarts due to a lack of memory or of CPU, xref:../../backup_and_restore/application_backup_and_restore/troubleshooting.adoc#oadp-pod-crash-resource-request[you can set specific resource requests for either of those resources].
+
+|`labels`
+|map [ link:https://pkg.go.dev/builtin#string[string] ] link:https://pkg.go.dev/builtin#string[string]
+|Labels to add to pods.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig[Complete schema definitions for the type `PodConfig`].
+
+.Features
+[options="header"]
+|===
+|Property|Type|Description
+
+|`dataMover`
+|*link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#DataMover[`DataMover`]
+|Defines the configuration of the Data Mover.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#Features[Complete schema definitions for the type `Features`].
+
+.DataMover
+[options="header"]
+|===
+|Property|Type|Description
+
+|`enable`
+|link:https://pkg.go.dev/builtin#bool[bool]
+|If set to `true`, deploys the volume snapshot mover controller and a modified CSI Data Mover plug-in. If set to `false`, these are not deployed.
+
+|`credentialName`
+|link:https://pkg.go.dev/builtin#string[string]
+|User-supplied Restic `Secret` name for Data Mover.
+
+|`timeout`
+|link:https://pkg.go.dev/builtin#string[string]
+|A user-supplied duration string for `VolumeSnapshotBackup` and `VolumeSnapshotRestore` to complete. Default is `10m` (10 minutes). A duration string is a possibly signed sequence of decimal numbers, each with optional fraction and a unit suffix, such as `300ms`, -1.5h` or `2h45m`. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, and `h`.
+|===
+
+The OADP API is more fully detailed in link:https://pkg.go.dev/github.com/openshift/oadp-operator[OADP Operator].
```

Comment:
I think these API docs need to be part of the OADP product (much like how the OCP API docs are part of the product).

These don't constitute as links to github (*but I don't like that we are relying on pkg.go.dev (and those docs) to document our API's vs these docs being on a Red Hat property.